client/src/components/UploadComponent.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from "react";
+import { v4 as uuidv4 } from "uuid";
+
+import { useHttpClient } from "../hooks/http-hook";
+
+const UploadComponent = ({ addVideo }) => {
+  const randomId = uuidv4();
+  const [videoData, setVideoData] = useState({
+    id: randomId,
```

Comment:
I'd generally expect the server to be generating IDs (because it can guarantee they're actually unique), rather than trusting the client to give it a unique value.